Controllers/Backend/Import.php

```diff
@@ -73,10 +73,19 @@ public function getImportedProductCategoriesTreeAction()
                 break;
             case strpos($parent, '_stream_') > 0:
                 list($shopId, $stream) = explode('_stream_', $parent);
-                $categories = $this->getCategoryExtractor()->getRemoteCategoriesTreeByStream($stream, $shopId, $hideMapped);
+                $categories = $this->getCategoryExtractor()->getRemoteCategoriesTreeByStream($stream, $shopId, $hideMapped, $shopId);
                 break;
             default:
-                $categories = $this->getCategoryExtractor()->getRemoteCategoriesTree($parent, false, $hideMapped);
+                $node = $this->request->getParam('id');
+                preg_match('/^(shopId(\d+)~)(.*)$/', $node, $matches);
```

Comment:
imho regexes should be documented what they are doing. For me it is always pain to debug and to find out why a regex was placed in a particular position. What do you think?